### DISCUSSION TITLE
RGPD cookie consent

### DISCUSSION THREAD

**Discussion by dunnno** [ArchiveBox/ArchiveBox#700] [Q&A]
Hi,

Is there any way to get rid of the RGPD consent box that appears above a web page and which gets archived ?

**Comment by pirate** [comment]
Definitely on our roadmap. It's going to be possible once we implement this ticket: https://github.com/ArchiveBox/ArchiveBox/issues/51

We have a design planned out, but it's still lots of work to get there though, don't expect it sooner than ~6mo.

**Reply by pirate** [reply]
For now I recommend using Chrome GUI to hide those prompts, then pass your CHROME_DATA_DIR to archivebox so that it has the cookie preferences saved in the profile. Hopefully it wont popup again during archiving if you've already closed them in that session. https://github.com/ArchiveBox/ArchiveBox/wiki/Configuration#chrome_user_data_dir

**Reply by dunnno** [reply]
Ok, but I don't understand if you're talking about my chrome browser installed on my puter or some chrome inside of ArchiveBox. Also, if the latter is the one you're talking about, how to reach the GUI of it ?

**Reply by pirate** [reply]
Both, the idea is to share your chrome profile directory between the browser on your computer and the one that archivebox uses. That way you close the cookie popups in your normal browser, quit, then load that profile into archivebox and it will have the popups hidden.

1. open your normal chrome on your computer and visit the sits you want to archive, close any popups / agree to cookies
2. quite chrome, and copy your entire https://chromium.googlesource.com/chromium/src/+/HEAD/docs/user_data_dir.md next to your archivebox data dir
3. if using docker, mount the copied data dir as a volume into the container, then run `archivebox config --set CHROME_DATA_DIR=/path/to/data/dir` and `archivebox --version` to check that it worked
4. `archivebox add 'https://example.com/some/test/url'` then check that there are no popups in the output

**Comment by dunnno** [comment]
Ok, i copied the whole Chrome folder inside the ArchiveBox directory "data". Ran the --set CHROME_USER_DATA_DIR command but I got an error : 
<img width="608" alt="Capture d’écran 2021-04-11 à 00 00 32" src="https://user-images.githubusercontent.com/51229129/114285662-234dcc00-9a59-11eb-8d00-47271ca95011.png">

There def is a Default folder with a bunch of stuff inside. I tried to set the path as /data/Chrome, ./Chrome, both with /Default at the end or not, no luck

EDIT : ok, my bad, I didn't mount it as a volume into the container.

EDIT again : well that didn't work, GDPR is still here :(

**Reply by pirate** [reply]
ah thats too bad, it usually works but maybe not on all sites. For particularly stubborn ones you'll probably have to wait until we implement JS scripting then.

**Reply by mwnoo** [reply]
For me this works when I set the `CHROME_DATA_DIR` to the copied chrome profile folder. But only when I change the permissions on the Cookie file to read-only. Otherwise for some reason the headless chrome browser removes most of the cookies. However, it only works for the screenshot, pdf and html output NOT for SingleFile which still shows the consent popups. BTW I only got this working with the apt installation of Archivebox on Linux. 

Unfortunately I can't get this to work in the Docker-compose version of Archivebox.
In the Docker-compose version the chrome profile folder is mounted,  `archivebox --version` shows a valid `CHROME_DATA_DIR` and with read-write permissions on the Cookie file again the headless browser removes most of the cookies. But when I change the permissions to read-only on the Cookie file it doesn't seem to work.....

**Comment by Klar** [comment]
any update on this one? is there an easy fix?

**Comment by pirate** [comment]
If you set up a chrome profile and accept/reject the banners with that profile, it'll save your choices and then when archiving the banners won't be shown. See the Chromium Install wiki for instructions.
